### PR TITLE
NPE with OTel when wrong HTTP protocol version is sent

### PR DESCRIPTION
**Pull request by brunobat**
Fix for #26149

### CODE REVIEW

File: integration-tests/opentelemetry/src/test/java/io/quarkus/it/opentelemetry/OpenTelemetryTestCase.java [hunk 1]
```diff
@@ -1,10 +1,12 @@
 package io.quarkus.it.opentelemetry;
 
-import static io.restassured.RestAssured.get;
-import static io.restassured.RestAssured.given;
+import static io.restassured.RestAssured.*;
```
Comment (gsmet) [inline integration-tests/opentelemetry/src/test/java/io/quarkus/it/opentelemetry/OpenTelemetryTestCase.java:3]: We avoid star imports in Quarkus.

File: integration-tests/opentelemetry/src/test/java/io/quarkus/it/opentelemetry/OpenTelemetryTestCase.java [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
 import static org.hamcrest.CoreMatchers.containsString;
 import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (gsmet) [inline integration-tests/opentelemetry/src/test/java/io/quarkus/it/opentelemetry/OpenTelemetryTestCase.java:6]: Same here.

File: extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/HttpInstrumenterVertxTracer.java [hunk 3]
```diff
@@ -154,16 +155,18 @@ private static class ServerAttributesExtractor implements HttpServerAttributesGe
...
-                        return null;
+                HttpVersion version = ((HttpServerRequest) request).version();
+                if (version != null) {
+                    switch (version) {
+                        case HTTP_1_0:
+                            return "1.0";
+                        case HTTP_1_1:
+                            return "1.1";
+                        case HTTP_2:
+                            return "2.0";
```
Comment (gsmet) [inline extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/HttpInstrumenterVertxTracer.java:166]: This looks a bit fragile. What happens if they add 2.1 at some point?

**Review by gsmet — COMMENTED** [review]
I added a couple of (naive) comments.

For bugfixes that we want to backport, please also add the `triage/backport?` label so that we include it in the next backport train.

File: integration-tests/opentelemetry/src/test/java/io/quarkus/it/opentelemetry/OpenTelemetryTestCase.java (hunk 1 above)
Comment (brunobat) [reply integration-tests/opentelemetry/src/test/java/io/quarkus/it/opentelemetry/OpenTelemetryTestCase.java:3]: will fix

File: integration-tests/opentelemetry/src/test/java/io/quarkus/it/opentelemetry/OpenTelemetryTestCase.java (hunk 2 above)
Comment (brunobat) [reply integration-tests/opentelemetry/src/test/java/io/quarkus/it/opentelemetry/OpenTelemetryTestCase.java:6]: will fix

File: extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/HttpInstrumenterVertxTracer.java (hunk 3 above)
Comment (brunobat) [reply extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/HttpInstrumenterVertxTracer.java:166]: Yeah, not the most extensible thing. This happens to comply with the HTTP version semantics required by OTel, here: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#common-attributes
This gets the version from Vert.x... Not sure of the best way to proceed here.

Comment (radcortez) [reply extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/HttpInstrumenterVertxTracer.java:166]: Yes, to follow the OTel naming conventions we need to convert the HTTP version format coming from Vert.x. We can probably be a bit smarter and use the value from Vert.x when we don't have a conversion available.

**Comment by radcortez** [comment]
Remember to build the changed code to apply the formatter: 

`mvn -pl extensions/opentelemetry/opentelemetry/runtime,extensions/opentelemetry/opentelemetry/deployment package`

File: extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/HttpInstrumenterVertxTracer.java (hunk 3 above)
Comment (brunobat) [reply extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/HttpInstrumenterVertxTracer.java:166]: sounds good to me, will update

**Review by radcortez — APPROVED** [review]

**Review by mamadaliev — APPROVED** [review]

**Comment by gsmet** [comment]
Thanks!

**Pull request merged**